src/main/java/seedu/address/logic/commands/SearchPatientCommand.java

```diff
@@ -0,0 +1,197 @@
+package seedu.address.logic.commands;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_NAME;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_TEMP_RANGE;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+import seedu.address.commons.core.Messages;
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.Model;
+import seedu.address.model.person.Name;
+import seedu.address.model.person.Person;
+import seedu.address.model.person.TemperatureRange;
+
+
+/**
+ * Search a person or a list of person according to a name or a range of temperature.
+ */
+public class SearchPatientCommand extends Command {
+
+    public static final String COMMAND_WORD = "searchpatient";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Search a patient in Covigent. "
+            + "Parameters: "
+            + PREFIX_NAME + "NAME || "
+            + PREFIX_TEMP_RANGE + "TEMPERATURERANGE "
+            + "Example: " + COMMAND_WORD + " "
+            + PREFIX_NAME + "John Doe ";
+
+    public static final String MESSAGE_SEARCH_PERSON_SUCCESS = "Person found: %1$s";
+    public static final String MESSAGE_NOT_FOUND = "At least one field to edit must be provided.";
+    public static final String MESSAGE_PERSON_NOT_FOUND = "The person you entered is not in the list.";
+    public static final String MESSAGE_SEARCH_PERSON_LIST_SUCCESS = "Person match your criteria found: \n";
+
+    private final SearchPatientDescriptor searchPatientDescriptor;
+    private final int areaNotFound = 0;
```

Comment:
Do you think an Enum may be preferable here since there are a limited number of possible values?

Using `int` is not very scalable. If we have 10 different search conditions, we would need 10 `int`, and it can be hard to keep track of these variables.

Perhaps we can declare a nested Enum:

```
private enum SearchCondition {
    SEARCH_BY_NAME.
    SEARCH_BY_TEMPERATURE,
    SEARCH_CONDITION_INVALID; // no condition or multiple conditions given
}
```

If you would like to use `int`, maybe it would be better to make them static and change the variable names to indicate they are constants. E.g.

```
private static final int SEARCH_BY_NAME = 1;
```

Personally, I think Enum might be neater. 😄

Reply:
This is really a good idea! Thanks for bringing it up